File: rust/tests/managed_agents_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_managed_agents_list_managed_agents_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .managed_agents
        .list_managed_agents(&"project_id".to_string(), None)
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/projects/project_id/managed-agents", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_managed_agents_create_managed_agent_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .managed_agents
        .create_managed_agent(
            &"project_id".to_string(),
            &CreateManagedAgentRequest {
                agent_slug: "agentSlug".to_string(),
                active_harness_version_id: None,
                compatibility_profile_id: None,
                display_name: None,
                harness_id: None,
                source: None,
                status: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count(
        "POST",
        "/v1/projects/project_id/managed-agents",
        None,
        1,
    )
    .await
    .unwrap();
}

File: ruby/test/wire/managed_agents_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class ManagedAgentsWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_managed_agents_list_managed_agents_with_wiremock
    test_id = "managed_agents.list_managed_agents.0"

    @client.managed_agents.list_managed_agents(
      project_id: "project_id",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "managed_agents.list_managed_agents.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/projects/project_id/managed-agents",
      query_params: nil,
      expected: 1
    )
  end

  def test_managed_agents_create_managed_agent_with_wiremock
    test_id = "managed_agents.create_managed_agent.0"

    @client.managed_agents.create_managed_agent(
      project_id: "project_id",
      agent_slug: "agentSlug",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "managed_agents.create_managed_agent.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/managed-agents",
      query_params: nil,
      expected: 1
    )
  end
end
